lib/pure/results.nim

```diff
@@ -0,0 +1,124 @@
+import typetraits, strutils
+
+type
+  Result*[T] = object
```

Comment:
This type is named `result<T, E>` in Rust, which was my inspiration. 

Any ideas for the name? `Either[A, ErrorDescription]` seems a bit too verbose. I don't know how to implement stack trace collection with just `Either[A, Exception]`, which is anyway too verbose.